### ISSUE TITLE
Map with timeline

### ISSUE THREAD

**Issue by joschne** [geovistory/design-system#116]
**Problem description**

The current [map plugin](https://design.geovistory.org/?path=/story/data-visualization-components-yasgui--with-map-plugin) is great for visualization of a static map.

Since Geovistory deals with historical data, time is an important aspect. In order to show developments of historical phenomena on a map, a time line would be helpful.   

**Functional requirement**
Starting on a query like the following, (based on [maritime history project](https://www.geovistory.org/project/84760/sparql)), the map should filter the records of a given time range.   

```sparql
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX ontome: <https://ontome.net/ontology/>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
PREFIX time: <http://www.w3.org/2006/time#>
   
SELECT ?s ?label ?long ?lat  ?event (MIN(?year) as ?minyear) (MAX(?year) as ?maxyear)
WHERE {

    # Geographical Place -had presence-> Presence -was at-> Place (lat/long)
    ?s ontome:p147i/ontome:p148 ?place.

    # Geographical Place -label-> label
    ?s rdfs:label ?label.

    # Geographical Place -is arrival place of-> Ship Voyage
    ?s ontome:p1335i ?event.
  	
    # Event has time span 
  	?event ontome:p4 ?timespan.
  
    # time span has date 
    ?timespan ?timeprop ?dtd.
    ?dtd a time:DateTimeDescription .
  	# date has year
  	?dtd time:year ?y .
    bind(xsd:integer(replace(str(?y), "^-", "" )) as ?year )
  	FILTER(isLiteral(?year))


    # Extract lat and long from WKT
    bind(replace(str(?place), '<http://www.opengis.net/def/crs/EPSG/0/4326>', "", "i") as ?rep)
    bind(xsd:float(replace(str(?rep), "^[^0-9\\.-]*([-]?[0-9\\.]+) .*$", "$1" )) as ?long )
    bind(xsd:float(replace( str(?rep), "^.* ([-]?[0-9\\.]+)[^0-9\\.]*$", "$1" )) as ?lat )

    # Append the project query param to the URI
    bind(concat(str(?s), "?p=84760") as ?link )
}
GROUP BY ?s ?label ?long ?lat ?event 
```

Show items without time.

**Technical Specification**

**Acceptance Criteria**

**Comment by joschne** [comment]
@flicksolutions please review specs

**Comment by flicksolutions** [comment]
@joschne 

> The map shows one circle per pointURI and type.
What does _and type_ mean here? Does it mean the possibility of the pointURI being in the list twice but with a different type and then two points are being displayed on top of each other?

> The size of a circle depends on the number of events
...at the same coordinates, right? It's not the case that all circles of EventA are the same size because there are 3 EventA in total.
On the same point, this kind of implies, that the size of the circles implies that the size depends on the number of different Events:
> one point for <geoplace1> with radius 2 (for Laura and May) and color A (for Work)
If that is the case, please make it explicit.

- I don't really get the difference between type and Event. Is a type something like a generalization of events? Isn't the fact that Maria worked in Lyon dependent on the Event? Why do we need a type?

> On init, the map and the timeline zoom to the extend of the data
timeline shouldn't _zoom_ right? The lowest value should just be the lowest value in the data and the max value should be the max value in the data and on init it should all be selected, right?

**Comment by joschne** [comment]
> What does _and type_ mean here? Does it mean the possibility of the pointURI being in the list twice but with a different type and then two points are being displayed on top of each other?

Exactly. In the example, `<geoplace2>` is displayed as two points, one per type.

**Comment by joschne** [comment]
> > The size of a circle depends on the number of events
> 
> ...at the same coordinates, right? It's not the case that all circles of EventA are the same size because there are 3 EventA in total. On the same point, this kind of implies, that the size of the circles implies that the size depends on the number of different Events:
> 
> > one point for  with radius 2 (for Laura and May) and color A (for Work)
> 
> If that is the case, please make it explicit.

Would this text make the specs more understandable?

The map displays one circle per pointURI and type. This means, we group the events per pointURI and type. Each group will be displayed as a circle. The circle radius is proportional to the number of events per group, the color depends on the type.

More technically, each group of pointURI and type aggregates the event attributes in a list: `start`, `end`, `eventLabel`, `eventURI` and picks one `pointLabel`, `lat`, `long` for each group. The output data structure could look like this (pseudo code): 
```json
{
  "<geoplace1>_Work": {
    "group": {
      "pointURI": "<geoplace1>",
      "pointLabel": "Bern",
      "type": "Work",
      "long": 12.3,
      "lat": 56.7
    },
    "events": [
      {
        "eventUri": "<eventA>",
        "eventLabel": "Laura in Bern",
        "start": "1597-02-25",
        "end": "1597-08-11"
      },
      {
        "eventUri": "<eventB>",
        ...
      }
    ]
  },
  "<geoplace2>_Work": {...}, 
  "<geoplace2>_Voyage": {...}
}
``` 

For each key, the map displays a circle. The color depends on group.type, the radius on events.length.
The above json example could also be expressed as an array of groups instead of an object with a key per group. I chose this approach, because it illustrates, that the groups identity is defined by pointUri and type.

**Comment by joschne** [comment]
> timeline shouldn't _zoom_ right? The lowest value should just be the lowest value in the data and the max value should be the max value in the data and on init it should all be selected, right?

correct!
